0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md

```diff
@@ -1,73 +1,56 @@
 
 ### Relation to Other Protocols
 
-SPSP is used for exchanging payment information before an ILP payment is initiated. The sender and receiver use the [STREAM](../0029-stream/0029-stream.md) transport protocol to generate the ILP packets. The receiver generates the shared secret and ILP address to be used in STREAM and communicates it to the sender over HTTPS.
-
-### Model of Operation
-
-We assume that the sender knows the receiver's SPSP endpoint (see [Payment Pointers](../0026-payment-pointers/0026-payment-pointers.md)).
-
-1. The sender's SPSP Client queries the receiver's SPSP Endpoint.
-2. The SPSP Endpoint responds with the receiver info, including the receiver's ILP address and the shared secret to be used in STREAM. It MAY respond with a balance associated with this SPSP receiver, i.e. in the case of an invoice.
-3. The sender constructs an ILP payment using the receiver's ILP address.
-4. The sender begins sending the payment.
-    1. The sender uses STREAM to establish a ILP/STREAM connection to the receiver.
-    2. The sender will construct a stream to the receiver on the ILP/STREAM connection.
-    3. The sender will adjust their `sendMax` to reflect the amount they're willing to send.
-    4. The receiver will adjust their `receiveMax` to reflect the amount they're willing to receive.
-    5. The sender's and receiver's STREAM modules will move as much value as possible while staying inside these bounds.
-    6. If the sender reaches their `sendMax`, they end the stream and the connection. If the receiver reaches their `receiveMax`, they will end the STREAM and the connection.
+SPSP is used for exchanging connection information before an ILP payment or data transfer is initiated. The SPSP Client and Server use the [STREAM](../0029-stream/0029-stream.md) transport protocol to generate the ILP packets. The SPSP Server generates the shared secret and ILP address to be used in STREAM and communicates it to the client over HTTPS.
 
 ## Specification
 
-The SPSP endpoint is a URI used by the sender to query information about the receiver (which may be an invoice) and set up payments. The SPSP endpoint URI MUST NOT contain query string parameters. The sender SHOULD treat the URI as opaque. There are several supported ways to refer to an SPSP endpoint:
+The SPSP Endpoint is a URL the payment pointer resolves to, used by the SPSP Client to query information about the SPSP Server and set up payments. SPSP Clients MUST NOT send query string parameters in requests to the SPSP Endpoint URL. SPSP Servers that receive query string parameters in an SPSP request MUST reject the request with a 400 Bad Request HTTP response code. SPSP Clients SHOULD treat the URL as opaque and not depend on any information they derive from the URL. There are several supported ways to refer to an SPSP Endpoint:
```

Comment:
The main reason was easier embedding in other content. If I put a payment pointer in this sentence does it have a query string, i.e. $twitter.xrptipbot.com/sharafian_? Seems to be less of a concern with the way we're using payment pointers in practice though.

I think it'd be fine to say query params are not allowed in the payment pointer spec but say that they are allowed in SPSP. You'd just have to use the URL form instead of the payment pointer form. I could see it being really useful for some sites using Web Monetization though.

Reply:
I'd leave them as not allowed unless there is a good reason to include them.

Payment Pointers are a sub-set of URLs. They don't have to include all of the same features.

Every feature we exclude is one less attack surface to deal with.

The purpose of Payment Pointers is a short easy to read and transcribe string that translates into the entry-point for a payment interaction. If you want a machine-readable string that is encoded as a QR code for example then use an actual URL.

Reply:
> I think it'd be fine to say query params are not allowed in the payment pointer spec but say that they are allowed in SPSP. You'd just have to use the URL form instead of the payment pointer form. I could see it being really useful for some sites using Web Monetization though.

@sharafian makes a good point here but I'd still recommend we don't allow query string params unless there is a need for them.

Do we foresee custom SPSP servers that accept special query params? That sounds like a recipe for fragmentation

Reply:
It just seems like a really odd decision to limit what kinds of URLs you can use. An SPSP URL is just a normal URL. Payment pointers define the opinionated way of expressing them but that's primarily intended for human-facing situations.

If I have a webserver that uses web monetization and I want to have a payment pointer for each user why should `https://myserver.example/spsp/ben123` be allowed but `https://myserver.example/spsp?user=ben123` be forbidden? they're just two different styles.